### DISCUSSION TITLE
Can't migrate to 5.5.37 from 5.4.10

### DISCUSSION THREAD

**Discussion by rruprai1** [unoplatform/uno#18656] [General]
Unable to update Uno.SDK from 5.4.10 to 5.5.37

After updating global.json to 5.5.37 Uno Platform outputs:
[INFO] The project does not contain the Uno.WinUI.DevServer nor Uno.UI.DevServer package.

And csproj file will not load, indicating the csproj can't be found.

**Comment by jeromelaban** [comment]
Thanks for the report. Which version of visual studio are you using? Does it still happen if you close visual studio and load the solution again?

**Comment by rruprai1** [comment]
Hi,

Using:
Microsoft Visual Studio Professional 2022 (64-bit) - LTSC 17.8
Version 17.8.10

Yes, did the IDE restart and a forced dotnet restore via command line. And that did not work. 

I am not seeing the 5.5.37 equivalent of:
INFO] Project information hasn't been loaded, retrying...
[INFO][13:45:46.30] Uno Remote Control initialized (5.4.38+Branch.release-stable-5.4.Sha.a0d9fa8496bdf9362a983aeb448bd161020110b6.a0d9fa8496bdf9362a983aeb448bd161020110b6)
[INFO] Loaded the Uno.WinUI.DevServer service (5.4.38, API v2).
[WARNING][13:45:50.70] Detected that the active framework was changed from/to WebAssembly/Desktop/Windows, reloading the project (See https://aka.platform.uno/singleproject-vs-reload)

**Reply by jeromelaban** [reply]
Thanks for the update. A this time, 17.8 is not supported by uno anymore, so you won't get the vsix update, you may want to update to 17.9 or later.

Still, you're saying it's not working, what you explain what is not working?

**Reply by rruprai1** [reply]
By "not working" is that the project file can't be loaded, it remains as a unloaded project, SLN and CSPROJ is in different folders also, if that matters. 

I was hoping this was like previous updates, just moving up the global value, restart the IDE and go. 

Switch VS versions is something I guess I can do, generally though, I was hoping LTSC versions of VS would have longer support with Uno. But what is UNO's policy for LTSC versions? 

It sounds like the problem is solely with VS2022 version, I have not tried VSCode to see if the same issue occurs.

**Reply by jeromelaban** [reply]
Got it. As the project is unloaded, if you right click on it then "reload", what happens?

As for the VS versions, because of net9 support we were forced to update the minimum to 17.9. We do not have an explicit policy on VS versions, but we try to stay compatible with older versions as long as possible.

**Reply by rruprai1** [reply]
"Reload" results in the same failure, unable to find the csproj file.  Which I guess means moving up VS versions :-)

**Reply by jeromelaban** [reply]
Can you show the exact error message? Also, the output window will show which error that is exactly.

**Reply by rruprai1** [reply]
Sorry for the delay, 

In VS2022 (v17.8.16) I see this:

error CS8032: An instance of analyzer Uno.Extensions.Generators.Analyzers.PropertySelectorAnalyzer cannot be created from uno.extensions.core\5.1.2\analyzers\dotnet\cs\Uno.Extensions.Core.Generators.dll : Could not load file or assembly 'System.Collections.Immutable, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified..

1>CSC : error CS9057: The analyzer assembly '\uno.ui.hotdesign\1.0.51\analyzers\dotnet\cs\Uno.UI.HotDesign.SourceGenerators.dll' references version '4.10.0.0' of the compiler, which is newer than the currently running version '4.8.0.0'.

1>CSC : error CS8032: An instance of analyzer Uno.Extensions.Generators.Analyzers.PropertySelectorAnalyzer cannot be created from uno.extensions.core\5.1.2\analyzers\dotnet\cs\Uno.Extensions.Core.Generators.dll : Could not load file or assembly 'System.Collections.Immutable, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified..

1>CSC : error CS9057: The analyzer assembly 'uno.ui.hotdesign\1.0.51\analyzers\dotnet\cs\Uno.UI.HotDesign.SourceGenerators.dll' references version '4.10.0.0' of the compiler, which is newer than the currently running version '4.8.0.0'.

In VSCode, none of the above happend. I guess that means I need to update to 17.10.X

**Reply by jeromelaban** [reply]
Thanks for the update. Note that VS 17.8 is not supported anymore by Uno, the minimum version is 17.10, which can explain the issue you're having. The version of visual studio that you are running does not include the minimum versions of the tooling that we expect.
